### PR TITLE
Match IA "Egg Utopia Launcher"

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: src/game/interactables_2/091.c
Comment (freshollie) [inline src/game/interactables_2/091.c:1]: Could name this file now?

Comment (JaceCear) [reply src/game/interactables_2/091.c:1]: On it!

File: src/game/interactables_2/091.c [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+            launcher->s.unk22 = 0x10;
+            launcher->s.focused = 0;
+
+            s->unk28->unk0 = -1;
+            s->unk10 = SPRITE_FLAG_PRIORITY(2);
+            s->graphics.dest = VramMalloc(EGG_UTO_LAUNCHER_TILE_COUNT);
+            s->graphics.anim = SA2_ANIM_LAUNCHER_EGG_UTO;
+            launcher->s.variant = 0;
+
+#ifndef NON_MATCHING
```
Comment (freshollie) [inline src/game/interactables_2/091.c:155]: Not sure this counts as a fake match? Typically nonmatching is used for that?

File: src/game/interactables_2/091.c [hunk 2]
```diff
@@ -0,0 +1,523 @@
...
+}
+
+static bool32 sub_807E044(Sprite_EggUtopia_Launcher *launcher)
+{
+    s16 posX, posY;
+
+    posX = launcher->posX - gCamera.x;
+    posY = launcher->posY - gCamera.y;
+
+    if (((posX + launcher->unk50) < -(CAM_REGION_WIDTH / 2))
```
Comment (freshollie) [inline src/game/interactables_2/091.c:427]: Does this fit any macros?

File: src/game/interactables_2/091.c [hunk 3]
```diff
@@ -0,0 +1,523 @@
...
+}
+
+static bool32 sub_807DDF0(Sprite_EggUtopia_Launcher *launcher)
+{
+    if (PLAYER_IS_ALIVE) {
+        s16 someX, someY;
+        s16 playerX, playerY;
+
+        // Launcher should only activate if it's the correct gravity
+        if (gUnknown_03005424 & EXTRA_STATE__GRAVITY_INVERTED) {
```
Comment (freshollie) [inline src/game/interactables_2/091.c]:
```suggestion
        if (GRAVITY_IS_INVERTED) {
```

File: src/game/interactables_2/091.c (hunk 1 above)
Comment (JaceCear) [reply src/game/interactables_2/091.c]: Well, it is a fake-match, but I didn't know how else to make it work.

Comment (freshollie) [reply src/game/interactables_2/091.c]: Is it fake? It's what they wrote

File: src/game/interactables_2/091.c (hunk 2 above)
Comment (JaceCear) [reply src/game/interactables_2/091.c]: Not *yet*. IS_OUT_OF_CAM_RANGE in game.h is closest, but we'd need a version supports top, bottom, left and right.
But in general they all have to be cleaned up and (hopefully) turned into a single or very few macros.

File: src/game/interactables_2/091.c (hunk 1 above)
Comment (freshollie) [reply src/game/interactables_2/091.c]: Maybe it shows there is some macro which made them non aware of it being redundant

File: include/sprite.h [hunk 4]
```diff
@@ -126,7 +126,6 @@ typedef struct {
 #define SPRITE_FLAG_X_FLIP(flip)                    ((flip) << 10)
 #define SPRITE_FLAG_Y_FLIP(flip)                    ((flip) << 11)
 #define SPRITE_FLAG_PRIORITY(prio)                  ((prio) << 12)
-#define SPRITE_FLAG_13(v)                           ((v) << 13)
```
Comment (freshollie) [inline include/sprite.h:129]: Ah, I think I added this? Good to know!

File: src/game/interactables_2/091.c (hunk 2 above)
Comment (JaceCear) [reply src/game/interactables_2/091.c]: For the record, that's also the case for small_propeller!

File: src/game/interactables_2/091.c (hunk 1 above)
Comment (JaceCear) [reply src/game/interactables_2/091.c]: OH, I thought you were talking about the register vars.
Github doesn't show the correct lines in the preview so I didn't realize what you were talking about.

No, I don't think that is a fake-match, but yes it's likely due to them using a macro.

**Review by freshollie — APPROVED** [review]

**Pull request merged**